sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala

```diff
@@ -145,7 +145,7 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       // For v2 ALTER TABLE statements, we have better error message saying view is not supported.
       assertAnalysisError(
         s"ALTER TABLE $viewName SET LOCATION '/path/to/your/lovely/heart'",
-        s"'$viewName' is a view not a table")
+        s"$viewName is a temp view not a table")
 
       // For the following v2 ALERT TABLE statements, unsupported operations are checked first
       // before resolving the relations.
```

Comment:
@cloud-fan This check (unsupported operations are checked first) seems inconsistent. For example, `DescribeRelation`'s unsupported operation is checked in `DataSourceV2Strategy.scala` after table resolution.

In this case, can we just fail with `Cannot alter a view with ALTER TABLE` and unsupported operations are checked in `DataSourceV2Strategy.scala` for consistency?

Reply:
you are right, this is inconsistent. +1 with your proposal.

Reply:
Thanks, I updated it.